### PR TITLE
Add allow_leading_plus and skip_white_space in chars_format

### LINKED ISSUE #275: Feature request: Add allow_leading_plus and skip_white_space in chars_format
* Remove/deprecate the defines `FASTFLOAT_ALLOWS_LEADING_PLUS` and `FASTFLOAT_SKIP_WHITE_SPACE`.
* Add `allow_leading_plus` and `skip_white_space` in `chars_format`.
* [Add an additional `from_chars_advanced` for integer types where the format options can be set.]

There may be different needs in different parts of the program using the fast_float library.

### PR DESCRIPTION
**Pull request by dalle**
* Add allow_leading_plus and skip_white_space in chars_format
* Fix fastfloat_strncasecmp and is_space for wchar_t and larger char types

Fixes #275

### CODE REVIEW

**Comment by dalle** [comment]
The PR may work against fixing #60

File: README.md [hunk 1]
```diff
@@ -444,9 +444,9 @@ https://github.com/fastfloat/fast_float/releases/download/v6.1.6/fast_float.h
 - It is part of the [brew package manager](https://formulae.brew.sh/formula/fast_float).
 - Some Linux distribution like Fedora include fast_float (e.g., as `fast_float-devel`).
 
-## RFC 7159
+## RFC 8259
```
Comment (lemire) [inline README.md:447]: This remark may be obsolete and was written before we had `fast_float::chars_format::json`. I wonder whether we couldn't not replace it with something like...

```
If you need support for RFC 8259 (JSON standard), you may want to consider use the `fast_float::chars_format::json` flag
```

Or something of the sort.

Thoughts?

Comment (dalle) [reply README.md:447]: Perhaps we can remove this section entirely, as `fast_float::chars_format::json` is exemplified above at line 283.

Comment (lemire) [reply README.md:447]: Yes. What about that?

I would not have made this comment, but given that you are modifying this section, maybe you can just delete it.

It is fine if you don't do it, and we can use another PR to do so.

Comment (dalle) [reply README.md:447]: I removed the section and updated the README.md with the new format flags as well, and some formatting changes.

**Comment by lemire** [comment]
merging

**Pull request merged**